packages/common/test/utils.spec.ts

```diff
@@ -57,122 +55,24 @@ describe('[Utils/Parse]', () => {
-        'tangerineWhistle',
-        'spuriousDragon',
-        'byzantium',
-        'constantinople',
-        'petersburg',
-        'istanbul',
-        'berlin',
-        'london',
-        'mergeForkIdTransition',
-        'paris',
-      ],
-      'hardfork parse order should be correct',
-    )
-    for (const hf of common.hardforks()) {
-      /* eslint-disable @typescript-eslint/no-use-before-define */
-      assert.equal(hf.forkHash, kilnForkHashes[hf.name], `${hf.name} forkHash should match`)
-    }
-
-    assert.equal(common.hardfork(), Hardfork.Paris, 'should correctly infer current hardfork')
-
-    // Ok lets schedule shanghai at block 0, this should force merge to be scheduled at just after
-    // genesis if even mergeForkIdTransition is not confirmed to be post merge
-    // This will also check if the forks are being correctly sorted based on block
-    Object.assign(gethGenesisKilnJSON.config, { shanghaiTime: Math.floor(Date.now() / 1000) })
-    const common1 = createCommonFromGethGenesis(gethGenesisKilnJSON, {
-      chain: 'customChain',
-    })
-    // merge hardfork is now scheduled just after shanghai even if mergeForkIdTransition is not confirmed
-    // to be post merge
-    assert.deepEqual(
-      common1.hardforks().map((hf) => hf.name),
-      [
-        'chainstart',
-        'homestead',
-        'tangerineWhistle',
-        'spuriousDragon',
-        'byzantium',
-        'constantinople',
-        'petersburg',
-        'istanbul',
-        'berlin',
-        'london',
-        'paris',
-        'mergeForkIdTransition',
-        'shanghai',
-      ],
-      'hardfork parse order should be correct',
-    )
-
-    assert.equal(common1.hardfork(), Hardfork.Shanghai, 'should correctly infer current hardfork')
+  it('should set merge to block 0 when terminalTotalDifficultyPassed is true', () => {
+    const mergeAtGenesisJson = {} as any
+    Object.assign(mergeAtGenesisJson, postMergeJSON)
+    mergeAtGenesisJson.config.terminalTotalDifficultyPassed = true
+    const common = createCommonFromGethGenesis(mergeAtGenesisJson, {})
+    assert.equal(common.hardforks().slice(-1)[0].block, 0)
   })
 
-  it('should successfully parse genesis with hardfork scheduled post merge', async () => {
-    const common = createCommonFromGethGenesis(postMergeHardforkJSON, {
-      chain: 'customChain',
-    })
-    assert.deepEqual(
-      common.hardforks().map((hf) => hf.name),
-      [
-        'chainstart',
-        'homestead',
-        'tangerineWhistle',
-        'spuriousDragon',
-        'byzantium',
-        'constantinople',
-        'petersburg',
-        'istanbul',
-        'muirGlacier',
-        'berlin',
-        'london',
-        'paris',
-        'shanghai',
-      ],
-      'hardfork parse order should be correct',
-    )
-
-    assert.equal(common.getHardforkBy({ blockNumber: 0n }), Hardfork.London, 'london at genesis')
-    assert.equal(
-      common.getHardforkBy({ blockNumber: 1n, td: 2n }),
```

Comment:
Same for this test, we can remove all checks with have to do with `td` checks, but the `getHardforkBy` via either block or timestamp should still work.

Reply:
See comment above